FormatXmlHelper constructor may throw IllegalArgumentException depending on the TransformerFactory impl

### Proposal

Something observed in my project when attempting to upgrade WireMock to 3.10. Depending on the `TransformerFactory` implementation in use, the constructor logic in `FormatXmlHelper` that attempts to set attributes on the factory may throw an `IllegalArgumentException`.

For example, I have one set of tests where `xalan` is on the classpath:

```
Caused by: java.lang.IllegalArgumentException: TransformerFactory does not recognise attribute 'http://javax.xml.XMLConstants/property/accessExternalDTD'.
	at org.apache.xalan.xsltc.trax.TransformerFactoryImpl.setAttribute(TransformerFactoryImpl.java:373)
	at org.apache.camel.quarkus.support.xalan.XalanTransformerFactory.setAttribute(XalanTransformerFactory.java:101)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.FormatXmlHelper.(FormatXmlHelper.java:69)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers$30.(WireMockHelpers.java:284)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers.(WireMockHelpers.java:283)
```

In some other tests, Saxon-HE is on the classpath:

```
Caused by: java.lang.IllegalArgumentException: Unrecognized configuration feature: indent-number
	at net.sf.saxon.Configuration.setConfigurationProperty(Configuration.java:4378)
	at net.sf.saxon.jaxp.SaxonTransformerFactory.setAttribute(SaxonTransformerFactory.java:340)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.FormatXmlHelper.(FormatXmlHelper.java:68)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers$30.(WireMockHelpers.java:284)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers.(WireMockHelpers.java:283)
```

Maybe `FormatXmlHelper` should gracefully handle this or is there any way to disable specific helpers?

### Reproduction steps

Create an empty application with the following dependencies:

```xml
<dependency>
  <groupId>org.wiremock</groupId>
  <artifactId>wiremock-standalone</artifactId>
  <version>3.10.0</version>
</dependency>
<dependency>
  <groupId>net.sf.saxon</groupId>
  <artifactId>Saxon-HE</artifactId>
  <version>12.5</version>
</dependency>
```

Then try to instantiate a `WireMockServer`:

```java
WireMockServer server = new WireMockServer();
```

It should result in:

```
Caused by: java.lang.IllegalArgumentException: Unrecognized configuration feature: indent-number
	at net.sf.saxon.Configuration.setConfigurationProperty(Configuration.java:4378)
	at net.sf.saxon.jaxp.SaxonTransformerFactory.setAttribute(SaxonTransformerFactory.java:340)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.FormatXmlHelper.<init>(FormatXmlHelper.java:68)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers$30.<init>(WireMockHelpers.java:284)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers.<clinit>(WireMockHelpers.java:283)
```

### References

_No response_

Temurin 21 here:

```java
Caused by: java.lang.IllegalArgumentException: Unrecognized configuration feature: indent-number
[INFO]  at net.sf.saxon.Configuration.setConfigurationProperty(Configuration.java:4373)
[INFO]  at net.sf.saxon.jaxp.SaxonTransformerFactory.setAttribute(SaxonTransformerFactory.java:340)
[INFO]  at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.FormatXmlHelper.<init>(FormatXmlHelper.java:6
8)
[INFO]  at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers$30.<init>(WireMockHelpers.jav
a:284)
[INFO]  at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers.<clinit>(WireMockHelpers.java
```

We have the same problem. In a project that uses `docx4j`, if we update WireMock to 3.10.0 we get the following stacktrace when initializing the WireMockServer:
```
java.lang.ExceptionInInitializerError
	at com.github.tomakehurst.wiremock.extension.responsetemplating.TemplateEngine.addHelpers(TemplateEngine.java:105)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.TemplateEngine.<init>(TemplateEngine.java:77)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.TemplateEngine.defaultTemplateEngine(TemplateEngine.java:56)
	at com.github.tomakehurst.wiremock.common.filemaker.FilenameMaker.<init>(FilenameMaker.java:44)
	at com.github.tomakehurst.wiremock.common.filemaker.FilenameMaker.<init>(FilenameMaker.java:40)
	at java.base/java.util.Objects.requireNonNullElseGet(Objects.java:314)
	at com.github.tomakehurst.wiremock.standalone.JsonFileMappingsSource.<init>(JsonFileMappingsSource.java:44)
	at com.github.tomakehurst.wiremock.core.WireMockConfiguration.getMappingsSource(WireMockConfiguration.java:157)
	at com.github.tomakehurst.wiremock.core.WireMockConfiguration.mappingsLoader(WireMockConfiguration.java:684)
	at com.github.tomakehurst.wiremock.core.WireMockApp.<init>(WireMockApp.java:96)
	at com.github.tomakehurst.wiremock.WireMockServer.<init>(WireMockServer.java:68)
	at com.myorg.app.fooprocessor.extentions.WireMockExtension.beforeAll(WireMockExtension.kt:28)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	Suppressed: kotlin.UninitializedPropertyAccessException: lateinit property mockServer has not been initialized
		at com.myorg.app.fooprocessor.extentions.WireMockExtension.afterAll(WireMockExtension.kt:19)
		... 1 more
Caused by: java.lang.IllegalArgumentException: Not supported: indent-number
	at org.docx4j.org.apache.xalan.processor.TransformerFactoryImpl.setAttribute(TransformerFactoryImpl.java:571)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.FormatXmlHelper.<init>(FormatXmlHelper.java:68)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers$30.<init>(WireMockHelpers.java:284)
	at com.github.tomakehurst.wiremock.extension.responsetemplating.helpers.WireMockHelpers.<clinit>(WireMockHelpers.java:283)
	... 13 more
```
Looking at the dependency tree it seems `docx4j` depends on `xalan`, so I don't think we can easily work around this issue.

--- a/src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/helpers/FormatXmlHelper.java
+++ b/src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/helpers/FormatXmlHelper.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2024 Thomas Akehurst
+ * Copyright (C) 2024-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -64,7 +64,7 @@ public class FormatXmlHelper extends AbstractFormattingHelper {
   private final TransformerFactory transformerFactory;
 
   public FormatXmlHelper() {
-    TransformerFactory factory = TransformerFactory.newInstance();
+    TransformerFactory factory = TransformerFactory.newDefaultInstance();
     factory.setAttribute("indent-number", 2);
     factory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
     factory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "");

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/FaultsTest.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/FaultsTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2024 Thomas Akehurst
+ * Copyright (C) 2024-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Http2AcceptanceTest.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Http2AcceptanceTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2019-2024 Thomas Akehurst
+ * Copyright (C) 2019-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Http2ClientFactory.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Http2ClientFactory.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2019-2024 Thomas Akehurst
+ * Copyright (C) 2019-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Http2DisabledAcceptanceTest.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Http2DisabledAcceptanceTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2024 Thomas Akehurst
+ * Copyright (C) 2024-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Jetty12MultipartParser.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Jetty12MultipartParser.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2018-2024 Thomas Akehurst
+ * Copyright (C) 2018-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Jetty12MultipartParserLoader.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/Jetty12MultipartParserLoader.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2018-2024 Thomas Akehurst
+ * Copyright (C) 2018-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/ProgrammaticHttpServerAcceptanceTest.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/ProgrammaticHttpServerAcceptanceTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2019-2024 Thomas Akehurst
+ * Copyright (C) 2019-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/server/CustomHttpServer.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/server/CustomHttpServer.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2024 Thomas Akehurst
+ * Copyright (C) 2024-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/server/CustomHttpServerFactory.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/server/CustomHttpServerFactory.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2024 Thomas Akehurst
+ * Copyright (C) 2024-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/servlet/AltHttpServerFactory.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/servlet/AltHttpServerFactory.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016-2024 Thomas Akehurst
+ * Copyright (C) 2016-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/servlet/AlternativeServletContainerTest.java
+++ b/wiremock-jetty12/src/test/java/com/github/tomakehurst/wiremock/jetty12/servlet/AlternativeServletContainerTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016-2024 Thomas Akehurst
+ * Copyright (C) 2016-2025 Thomas Akehurst
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

